BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsModule/DiagnosticsEventListener.cs

```diff
@@ -81,5 +80,39 @@ protected override void OnEventSourceCreated(EventSource eventSource)
 
             base.OnEventSourceCreated(eventSource);
         }
+
+        /// <summary>
+        /// This method checks if the given EventSource Name matches known EventSources that we want to subscribe to.
+        /// </summary>
+        private static bool ShouldSubscribe(EventSource eventSource)
+        {
+            if (eventSource.Name.StartsWith("Microsoft-A", StringComparison.Ordinal))
+            {
+                switch (eventSource.Name)
+                {
+                    case "Microsoft-ApplicationInsights-Core": 
+                    case "Microsoft-ApplicationInsights-Data": 
+                    case "Microsoft-ApplicationInsights-WindowsServer-TelemetryChannel": 
+                    case "Microsoft-ApplicationInsights-Extensibility-AppMapCorrelation-Dependency":
+                    case "Microsoft-ApplicationInsights-Extensibility-AppMapCorrelation-Web":
+                    case "Microsoft-ApplicationInsights-Extensibility-DependencyCollector":
+                    case "Microsoft-ApplicationInsights-Extensibility-EventCounterCollector":
+                    case "Microsoft-ApplicationInsights-Extensibility-HostingStartup":
+                    case "Microsoft-ApplicationInsights-Extensibility-PerformanceCollector":
+                    case "Microsoft-ApplicationInsights-Extensibility-PerformanceCollector-QuickPulse":
+                    case "Microsoft-ApplicationInsights-Extensibility-Web":
+                    case "Microsoft-ApplicationInsights-Extensibility-WindowsServer":
+                    case "Microsoft-ApplicationInsights-WindowsServer-Core":
+                    case "Microsoft-ApplicationInsights-Extensibility-EventSourceListener":
+                    case "Microsoft-ApplicationInsights-AspNetCore":
+                    case "Microsoft-AspNet-Telemetry-Correlation": // https://github.com/aspnet/Microsoft.AspNet.TelemetryCorrelation/blob/master/src/Microsoft.AspNet.TelemetryCorrelation/AspNetTelemetryCorrelationEventSource.cs
```

Comment:
@rajkumar-rangaraj added new one for ilogger. We need to add that too here.

Reply:
Done. "Microsoft-ApplicationInsights-LoggerProvider"